stackslib/src/core/mempool.rs

```diff
@@ -316,15 +318,16 @@ impl MemPoolAdmitter {
         self.cur_consensus_hash = cur_consensus_hash.clone();
         self.cur_block = cur_block.clone();
     }
-    pub fn will_admit_tx<Conn>(
```

Comment:
So, one wrinkle with adding all these type parameters is that a copy of this code will be generated for each type concretization. This would bloat the binary considerably and would slow down compile times.

Reply:
Yeah, that's the downside, but compile-time safety and a little performance would be to gain. I'm sort-of counting on that there are only three implementations (null, memory and sqlite), and we could gate null + memory to only compile for `test`, so my hope is that it shouldn't bloat _too_ much.